lib/std/os/bits/linux/sparc64.zig

```diff
@@ -0,0 +1,617 @@
+    shmctl = 396,
+    shmat = 397,
+    shmdt = 398,
+    msgget = 399,
+    msgsnd = 400,
+    msgrcv = 401,
+    msgctl = 402,
+    pidfd_send_signal = 424,
+    io_uring_setup = 425,
+    io_uring_enter = 426,
+    io_uring_register = 427,
+    open_tree = 428,
+    move_mount = 429,
+    fsopen = 430,
+    fsconfig = 431,
+    fsmount = 432,
+    fspick = 433,
+    pidfd_open = 434,
+    openat2 = 437,
+    pidfd_getfd = 438,
+
+    _,
+};
+
+pub const O_CREAT = 0x200;
+pub const O_EXCL = 0x800;
+pub const O_NOCTTY = 0x8000;
+pub const O_TRUNC = 0x400;
+pub const O_APPEND = 0x8;
+pub const O_NONBLOCK = 0x4000;
+pub const O_SYNC = 0x802000;
+pub const O_DSYNC = 0x2000;
+pub const O_RSYNC = O_SYNC;
+pub const O_DIRECTORY = 0x10000;
+pub const O_NOFOLLOW = 0x20000;
+pub const O_CLOEXEC = 0x400000;
+
+pub const O_ASYNC = 0x40;
+pub const O_DIRECT = 0x100000;
+pub const O_LARGEFILE = 0;
+pub const O_NOATIME = 0x200000;
+pub const O_PATH = 0x1000000;
+pub const O_TMPFILE = 0x2010000;
+pub const O_NDELAY = O_NONBLOCK | 0x4;
+
+pub const F_DUPFD = 0;
+pub const F_GETFD = 1;
+pub const F_SETFD = 2;
+pub const F_GETFL = 3;
+pub const F_SETFL = 4;
+
+pub const F_SETOWN = 5;
+pub const F_GETOWN = 6;
+pub const F_GETLK = 7;
+pub const F_SETLK = 8;
+pub const F_SETLKW = 9;
+
+pub const F_RDLCK = 1;
+pub const F_WRLCK = 2;
+pub const F_UNLCK = 3;
+
+pub const F_SETOWN_EX = 15;
+pub const F_GETOWN_EX = 16;
+
+pub const F_GETOWNER_UIDS = 17;
+
+pub const LOCK_SH = 1;
+pub const LOCK_EX = 2;
+pub const LOCK_NB = 4;
+pub const LOCK_UN = 8;
+
+/// stack-like segment
+pub const MAP_GROWSDOWN = 0x0200;
+
+/// ETXTBSY
+pub const MAP_DENYWRITE = 0x0800;
+
+/// mark it as an executable
+pub const MAP_EXECUTABLE = 0x1000;
+
+/// pages are locked
+pub const MAP_LOCKED = 0x0100;
+
+/// don't check for reservations
+pub const MAP_NORESERVE = 0x0040;
+
+pub const VDSO_CGT_SYM = "__vdso_clock_gettime";
+pub const VDSO_CGT_VER = "LINUX_2.6";
+
+pub const Flock = extern struct {
+    l_type: i16,
+    l_whence: i16,
+    l_start: off_t,
+    l_len: off_t,
+    l_pid: pid_t,
+};
+
+pub const msghdr = extern struct {
+    msg_name: ?*sockaddr,
+    msg_namelen: socklen_t,
+    msg_iov: [*]iovec,
+    msg_iovlen: u64,
+    msg_control: ?*c_void,
+    msg_controllen: u64,
+    msg_flags: i32,
+};
+
+pub const msghdr_const = extern struct {
+    msg_name: ?*const sockaddr,
+    msg_namelen: socklen_t,
+    msg_iov: [*]iovec_const,
+    msg_iovlen: u64,
+    msg_control: ?*c_void,
+    msg_controllen: u64,
+    msg_flags: i32,
+};
+
+pub const off_t = i64;
+pub const ino_t = u64;
+
+/// Renamed to Stat to not conflict with the stat function.
+/// atime, mtime, and ctime have functions to return `timespec`,
+/// because although this is a POSIX API, the layout and names of
+/// the structs are inconsistent across operating systems, and
+/// in C, macros are used to hide the differences. Here we use
+/// methods to accomplish this.
+pub const Stat = extern struct {
```

Comment:
That looks bad, however, I'm not sure how to proceed with this.
Should I make a new struct for the kernel's version of stat so that the library code can use it when it's finally get decoupled, or should I do something else?

Reply:
I'll try to fix this mess asap in the next few days.
If you just want to get the port going I'd use something like this:
```zig
const kernel_stat = extern struct {
// stat according to the kernel...
};
const libc_stat = extern struct {
// stat64 according to the libc
};
const Stat = kernel_stat;
```
The unused stat definition will come in handy later.

Reply:
Okay, added a separate kernel_stat struct.